py/server/deephaven/table.py

```diff
@@ -407,19 +408,82 @@ def _sort_column(col, dir_):
         _JColumnName.of(col)))
 
 
-def _td_to_columns(table_definition):
-    cols = []
-    j_cols = table_definition.getColumnsArray()
-    for j_col in j_cols:
-        cols.append(
-            Column(
-                name=j_col.getName(),
-                data_type=dtypes.from_jtype(j_col.getDataType()),
-                component_type=dtypes.from_jtype(j_col.getComponentType()),
-                column_type=ColumnType(j_col.getColumnType()),
-            )
+TableDefinitionAlias = Union[Mapping[str, dtypes.DType], Iterable[ColumnDefinition], 'TableDefinition']
+"""The TableDefinition type alias"""
+
+class TableDefinition(JObjectWrapper, Mapping):
+    """A Deephaven table definition, as a mapping from column name to ColumnDefinition."""
+
+    j_object_type = _JTableDefinition
+
+    @staticmethod
+    def of(table_definition: TableDefinitionAlias) -> 'TableDefinition':
```

Comment:
See other comment on `of` being Java idiomatic, but `from` being python idiomatic.

Reply:
Can "from" be directly used as method name in python? At least not directly, right, b/c `from X import Y` syntax, right?